subsys/net/lib/sockets/sockets_packet.c

```diff
@@ -386,7 +386,8 @@ static const struct socket_op_vtable packet_sock_fd_op_vtable = {
 static bool packet_is_supported(int family, int type, int proto)
 {
 	if (((type == SOCK_RAW) && (proto == ETH_P_ALL)) ||
-		((type == SOCK_RAW) && (proto == IPPROTO_RAW)) ||
+	    ((type == SOCK_RAW) && (proto == IPPROTO_RAW)) ||
+	    ((type == SOCK_RAW) && (proto == ETH_P_ECAT)) ||
```

Comment:
This change should be introduced after we have something that can receive EtherCAT frames. Do you have plans to submit the actual EtherCAT stack implementation?

Reply:
The EtherCAT frames are filled in application layer, then use socket (domain: PF_PACKET and type: SOCK_RAW) to pass them to ethernet driver to send out without changing; the reverse process in receive direction. So nothing is needed in Zephyr networking stack but check the protocol, and this patch is just to let the socket interface permit this kind of packets pass through.

No plan, currently I'm using SOEM, but wondering if it can be integrated into Zephyr.

Thanks,
Zhiqiang

Reply:
Ok, makes sense.